src/lsm/schema.zig

```diff
@@ -458,9 +460,29 @@ pub const Manifest = struct {
         assert(context.entry_count == @divExact(header.size - @sizeOf(vsr.Header), entry_size));
         assert(stdx.zeroed(&context.reserved));
 
+        if (header.commit == 0) {
+            assert(header.parent == 0);
+        }
+
         return .{ .entry_count = context.entry_count };
     }
 
+    pub fn previous(manifest_block: BlockPtrConst) ?struct { checksum: u128, address: u64 } {
+        _ = from(manifest_block); // Validation only.
+
+        const header = header_from_block(manifest_block);
+        if (header.commit == 0) {
+            assert(header.parent == 0);
+
+            return null;
+        } else {
+            return .{
+                .checksum = header.parent,
+                .address = header.commit,
+            };
+        }
+    }
```

Comment:
Mostly unrelated, but maybe call this whole struct `schema.ManifestNode`, to not confuse with actual manifest?